sdcm/utils/common.py

```diff
@@ -911,6 +912,81 @@ def clean_test_security_groups(tags_dict, dry_run=False):
+    :param region_name: name of the region to list
+    :param group_as_region: if True the results would be grouped into regions
+    :param verbose: if True will log progress information
+
+    :return: load balancers dict where region is a key
+    """
+    load_balancers = {}
+    aws_regions = [region_name] if region_name else all_aws_regions()
+
+    def get_load_balancers(region):
+        if verbose:
+            LOGGER.info('Going to list aws region "%s"', region)
+        time.sleep(random.random())
+        tagging = boto3.client('resourcegroupstaggingapi', region_name=region)
+        paginator = tagging.get_paginator('get_resources')
+        tag_filter = [{'Key': key, 'Values': [value]}
+                      for key, value in tags_dict.items() if key != 'NodeType']
+
+        tag_mappings = itertools.chain.from_iterable(
+            page['ResourceTagMappingList']
+            for page in paginator.paginate(TagFilters=tag_filter, ResourceTypeFilters=['elasticloadbalancing:loadbalancer'])
+        )
+        load_balancers[region] = list(tag_mappings)
+        if verbose:
+            LOGGER.info("%s: done [%s/%s]", region, len(list(load_balancers.keys())), len(aws_regions))
+
+    ParallelObject(aws_regions, timeout=100).run(get_load_balancers, ignore_exceptions=False)
+
+    if not group_as_region:
+        load_balancers = list(itertools.chain(*list(load_balancers.values())))  # flatten the list of lists
+        total_items = load_balancers
+    else:
+        total_items = sum([len(value) for _, value in load_balancers.items()])
+    if verbose:
+        LOGGER.info("Found total of %s ips.", total_items)
+    return load_balancers
+
+
+def clean_load_balancers_aws(tags_dict, dry_run=False):
+    """
+    Remove all load balancers with specific tags AWS
+
+    :param tags_dict: a dict of the tag to select the groups, e.x. {"TestId": "9bc6879f-b1ef-47e1-99ab-020810aedbcc"}
+    :param dry_run: if True, wouldn't delete any resource
+    :return: None
+    """
+
+    # don't if `post_behavior_k8s_cluster` was set to not destroy
+    if "NodeType" in tags_dict and tags_dict.get("NodeType") != "k8s":
+        return
+
+    assert tags_dict, "tags_dict not provided (can't clean all instances)"
+    aws_instances = list_load_balancers_aws(tags_dict=tags_dict, group_as_region=True)
+
+    for region, elb_list in aws_instances.items():
+        if not elb_list:
+            LOGGER.info("There are no ELBs to remove in AWS region %s", region)
+            continue
+        client = boto3.client('elb', region_name=region)
+        for elb in elb_list:
+            arn = elb.get('ResourceARN')
+            LOGGER.info("Going to delete '%s'", arn)
+            if not dry_run:
+                try:
+                    response = client.delete_load_balancer(LoadBalancerName=arn.split('/')[1])
+                    LOGGER.debug("Done. Result: %s\n", response)
```

Comment:
Maybe better to make all such calls be parallel/concurrent?
None of load balancers depends on any other.

Reply:
since we have only one per test (for now, later we'll have one per region), it's doesn't make a lot of difference.

Reply:
How fast is it? If seconds, then ok.

Reply:
we I didn't delete a lot of those, main one or two, and yest it was quick (it's not like deleting a node pool or eks cluster)